browse/controllers/stats_page/__init__.py

```diff
@@ -0,0 +1,144 @@
+"""Handle requests to display and return stats about the arXiv service."""
+
+import dateutil.parser
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, Tuple
+from werkzeug.exceptions import InternalServerError, BadRequest
+
+from arxiv import status
+from arxiv.base import logging
+from browse.services.database import get_hourly_stats, get_hourly_stats_count, \
+    get_monthly_download_stats, get_monthly_submission_stats, \
+    get_monthly_submission_count, get_monthly_download_count, \
+    get_max_download_stats_dt
+from browse.services.document.config.deleted_papers import DELETED_PAPERS
+
+
+Response = Tuple[Dict[str, Any], int, Dict[str, Any]]
+logger = logging.getLogger(__name__)
+
+
+def get_hourly_stats_page(requested_date_str: Optional[str] = None) -> Response:
+    """Get data for the /stats/today page."""
+    response_data: Dict[str, Any] = {}
+    try:
+        current_dt = datetime.now()
+        requested_dt = current_dt - timedelta(hours=1)
+        if requested_date_str:
+            requested_dt = dateutil.parser.parse(requested_date_str)
+        normal_count, admin_count, num_nodes = \
+            get_hourly_stats_count(stats_date=requested_dt.date())
+
+        response_data['current_dt'] = current_dt
+        response_data['requested_dt'] = requested_dt
+        response_data['normal_count'] = normal_count
+        response_data['admin_count'] = admin_count
+        response_data['num_nodes'] = num_nodes
+        return response_data, status.HTTP_200_OK, {}
+    except (TypeError, ValueError):
```

Comment:
Again, big ``try`` blocks -- it makes it a little harder to see where you expect the problem to arise, and what the nature of these exceptions might be. And don't forget chaining.